Minor bugfix for signArtifacts.groovy

abhinavGupta16 opened the pull request:
Signed-off-by: Abhinav Gupta <abhng@amazon.com>

### Description
Currently, the pgp key for signing is only imported if the file is downloaded. We only need to download the file ones, however, we need to import the key if the file already exists or when we download the file.
 
### Issues Resolved
pgp key failure will not happen for preexisting pgp key
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

peternied commented on vars/signArtifacts.groovy:
```diff
@@ -30,8 +30,9 @@ void call(Map args = [:]) {
 
 void importPGPKey(){
+    keyPath = "$WORKSPACE/opensearch.pgp"
     if( !fileExists("$WORKSPACE/opensearch.pgp")) {
```
this if statement should use `keypath`

peternied reviewed (changes requested):
Since this was a bug, lets add unit tests to confirm it works as expected

zelinh commented on vars/signArtifacts.groovy:
```diff
@@ -30,8 +30,9 @@ void call(Map args = [:]) {
 }
 
 void importPGPKey(){
+    keyPath = "$WORKSPACE/opensearch.pgp"
```
Is this `keyPath` necessary? I think we are justing taking the `gpg import` out of the if bracket.

abhinavGupta16 replied:
Yeah, I realized that the same path is being used at multiple places. So it is better to have in a variable. This makes it less error prone (for any changes in the path in future) and easy to follow.

abhinavGupta16 commented:
> Since this was a bug, lets add unit tests to confirm it works as expected

I have added a test case for it. I have also changed the test case to be a unit test case only `signArtifacts` lib and added another test case for the actual job. I believe we should do this for every jenkins job we have in the project.

dblock commented on vars/signArtifacts.groovy:
```diff
@@ -30,8 +30,9 @@ void call(Map args = [:]) {
-        sh("gpg --import $WORKSPACE/opensearch.pgp")
+    keyPath = "$WORKSPACE/opensearch.pgp"
+    if( !fileExists("${keyPath}")) {
+        sh("curl -SL https://artifacts.opensearch.org/publickeys/opensearch.pgp -o ${keyPath}")
```
Since you import every time, how about never storing this key?
https://www.reddit.com/r/GnuPG/comments/4oz7zd/possible_to_import_gpg_key_from_a_url/

```
curl -sSL http://my.key.url | gpg --import -
```

peternied commented on jenkins/sign-artifacts/sign-standalone-artifacts.jenkinsfile.txt:
```diff
@@ -0,0 +1,33 @@
+   sign-standalone-artifacts.run()
```
Please move this file into the `.../tests/data` path with the other files

abhinavGupta16 replied:
data was being used to store dummy files that are required for tests. I have created a separate folder to hold regression txt files for jenkins job, `tests/jenkins/jenkinsjob-regression-files`. Let me know if the name of the dir works

peternied reviewed (approved):
Thanks for updating the path on those tests, looks good
